### PR TITLE
Allow returning to be used with oracle update

### PR DESCRIPTION
**Pull request by AiNoKame**
https://github.com/tgriesser/knex/issues/1249

### CODE REVIEW

File: src/dialects/oracle/query/compiler.js [hunk 1]
```diff
@@ -105,9 +105,19 @@ assign(QueryCompiler_Oracle.prototype, {
   update: function() {
     var updates = this._prepUpdate(this.single.update);
     var where   = this.where();
-    return 'update ' + this.tableName +
-      ' set ' + updates.join(', ') +
-      (where ? ' ' + where : '');
+    var returning = this.single.returning;
+
+    if (!returning) {
+      return 'update ' + this.tableName + ' set ' + updates.join(', ') + (where ? ' ' + where : '');
```
Comment (wubzz) [inline src/dialects/oracle/query/compiler.js]: @AiNoKame Would you mind declaring this as a seperate variable `sql` and using said variable here and in the call to `_addReturningToSqlAndConvert` ?

Not too fond of using the same string concatenation more than once in a function. Also would make this easier to read.

Comment (AiNoKame) [inline src/dialects/oracle/query/compiler.js]: Sure thing!

Comment (AiNoKame) [inline src/dialects/oracle/query/compiler.js]: Amended :smile:

Comment (AiNoKame) [inline src/dialects/oracle/query/compiler.js]: @wubzz Amended once more with line breaks in the sql statement to pass all the travis build tests

File: test/unit/query/builder.js [hunk 2]
```diff
@@ -2238,6 +2238,26 @@ describe("QueryBuilder", function() {
...
+    testsql(qb().from('users').where('id', '=', 1).update({email: 'foo', name: 'bar'}, '*'), {
+      oracle: {
+        sql: 'update "users" set "email" = ?, "name" = ? where "id" = ? returning ROWID into ?',
+        // bindings: ['foo', 'bar', 1]
+        bindings: function(bindings) {
+          expect(bindings.length).to.equal(4);
+          expect(bindings[0]).to.equal('foo');
+          expect(bindings[1]).to.equal('bar');
+          expect(bindings[2]).to.equal(1);
+          expect(bindings[3].toString()).to.equal('[object ReturningHelper:*]');
```
Comment (rhys-vdw) [inline test/unit/query/builder.js]: What's going on here? Not all that familiar with Oracle SQL.

**Comment by rhys-vdw** [comment]
Could you please update the docs to show that `returning` is supported by Oracle also?

File: test/unit/query/builder.js (hunk 2 above)
Comment (AiNoKame) [inline test/unit/query/builder.js]: Oh, I should get rid of that comment on 2245. The bindings for `returning ROWID into ?` should link to the OutParam obj used returningSql helper (following the pattern for the insert method returning tests https://github.com/tgriesser/knex/blob/master/test/unit/query/builder.js#L2329)

Comment (AiNoKame) [inline test/unit/query/builder.js]: Basically, the update normally just returns the ROWID, and the returningSql helper uses that ROWID in a select query/subquery directly after the update

**Comment by AiNoKame** [comment]
@wubzz Absolutely! I updated the `index.html`

**Pull request merged**